data/transactions/logic/assembler.go

```diff
@@ -760,6 +764,10 @@ func assembleGtxn(ops *OpStream, spec *OpSpec, args []string) error {
 	if !ok {
 		return fmt.Errorf("gtxn unknown arg %s", args[1])
 	}
+	_, ok = txnaFieldSpecByField[fs.field]
+	if ok {
+		return fmt.Errorf("gtxn unknown arg %s", args[1])
```

Comment:
two requests:
1. could you make the error string here differ than the one above ?
2. are you sure you need to print the arg[1] here in the error ? I'm asking because you didn't use it as a qualifier to determine that there was an error. ( i.e. you used fs.field, which you want to print out )

Reply:
I think here and in assembleTxn we know a little bit more, it matches a list, it is a known thing but in the wrong place? maybe we can note that in the error message?

Reply:
My intention was in preserving the same error as in original TEAL v1. If you think it is not important I'll change to a better one.

Reply:
I think that it's important to preserve the TEAL v1 language support, but it's ok to add more ( and better ! ) error messages for invalid TEAL programs. ( i.e. we've creating a better compiler )

Reply:
Fixed!
1. Use `found (g)txna field in (g)txn op` message
2. Update tests

> are you sure you need to print the arg[1] here in the error 

`arg[1]` here is the same as `fs.field.String()`, plus, we want to report back to the user exact string from his input